ssp

```diff
@@ -4853,59 +4731,29 @@ sub check_for_cpsources_conf {
                 push @cpsources_servers, $value;
             }
         }
-
-      CPSOURCES_SERVER:
         foreach my $cpsources_server (@cpsources_servers) {
             chomp $cpsources_server;
             my ( $host, $port ) = split( ':', $cpsources_server );
             if ( !defined($port) ) {
                 $port = 80;
             }
-
-            local $SIG{'ALRM'} = sub {
-                print_warn('/etc/cpsources.conf: ');
-                print_warning("Unresponsive mirror: http://$host:$port");
-                next CPSOURCES_SERVER;
-            };
-
-            my $sock = IO::Socket::INET->new(
-                PeerAddr => $host,
-                PeerPort => $port,
-                Proto    => 'tcp',
-                Timeout  => 3,
-            );
-
-            my $tiers_data;
-
-            if ($sock) {
-                print $sock "GET /cpanelsync/TIERS HTTP/1.1\r\nHost: $host\r\n\r\n";
-                sysread $sock, $tiers_data, 1000;
-                close $sock;
-            }
-
-            alarm 0;
-
+            my $tiers_data = _http_get( Host => $host, Port => $port, Path => '/cpanelsync/TIERS', WantHeaders => 1 );
             if ( !$tiers_data ) {
                 print_warn('/etc/cpsources.conf: ');
                 print_warning("Unresponsive mirror: http://$host:$port");
-                next CPSOURCES_SERVER;
+                next;
             }
-
-            # We care about the first line.
-            # HTTP/1.1 200 OK
-
             ( my $http_response ) = split( '\n', $tiers_data );
             ($http_response) = split( '\r', $http_response );
-
             if ( $http_response =~ /^HTTP\/(\d*?)\.(\d*?) (\d\d\d) (.*?)$/ ) {
                 if ( $3 != 200 ) {
                     print_warn('/etc/cpsources.conf: ');
-                    print_warning("Server $host:$port responds with $http_response when grabbing TIERS file.");
+                    print_warning("Server $host:$port responded with [ $http_response ] when fetching TIERS file.");
```

Comment:
Why not drop /^Servers\s// completely?  the $host:$port readout should be self explanatory